===================================================================
RCS file: /cvs/src/src/gdb/ada-tasks.c,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -r1.7 -r1.8
--- gdb/ada-tasks.c	2003/06/17 20:58:32	1.7
+++ gdb/ada-tasks.c	2004/04/28 16:36:25	1.8
@@ -4,7 +4,7 @@
 .
    This file is part of GDB.
 
-   [$Id$]
+   [$Id$]
    Authors: Roch-Alexandre Nomine Beguin, Arnaud Charlet <charlet@gnat.com>
 
    This program is free software; you can redistribute it and/or modify
@@ -794,10 +794,8 @@
     {
       /* FIXME: find_printable_frame should be defined in frame.h, and
          implemented in ada-lang.c */
-      /*      find_printable_frame (deprecated_selected_frame, frame_relative_level (deprecated_selected_frame)); */
       printf_filtered ("[Switching to task %d]\n", num);
-      print_stack_frame (deprecated_selected_frame,
-			 frame_relative_level (deprecated_selected_frame), 1);
+      print_stack_frame (get_selected_frame (), 1, SRC_AND_LOC);
     }
   else
     printf_filtered ("Unable to switch to task %d\n", num);
